src/Microsoft.AspNetCore.WebHooks.Receivers/Filters/WebHookSecurityFilter.cs

```diff
@@ -149,80 +161,124 @@ protected virtual IActionResult EnsureSecureConnection(string receiverName, Http
         /// Gets the locally configured WebHook secret key used to validate any signature header provided in a WebHook
         /// request.
         /// </summary>
-        /// <param name="request">The current <see cref="HttpRequest"/>.</param>
+        /// <param name="sectionKey">
+        /// The key (relative to <see cref="WebHookConstants.ReceiverConfigurationSectionKey"/>) of the
+        /// <see cref="IConfigurationSection"/> containing the receiver-specific
+        /// <see cref="WebHookConstants.SecretKeyConfigurationKeySectionKey"/> <see cref="IConfigurationSection"/>.
+        /// Typically this is the name of the receiver e.g. <c>github</c>.
+        /// </param>
         /// <param name="routeData">
         /// The <see cref="RouteData"/> for this request. A (potentially empty) ID value in this data allows a
         /// <see cref="WebHookSecurityFilter"/> subclass to support multiple senders with individual configurations.
         /// </param>
-        /// <param name="configurationName">
-        /// The name of the configuration to obtain. Typically this the name of the receiver, e.g. <c>github</c>.
-        /// </param>
         /// <param name="minLength">The minimum length of the key value.</param>
         /// <param name="maxLength">The maximum length of the key value.</param>
-        /// <returns>A <see cref="Task"/> that on completion provides the configured WebHook secret key.</returns>
-        [SuppressMessage("Microsoft.Reliability", "CA2000:Dispose objects before losing scope", Justification = "Disposed by caller")]
-        protected async virtual Task<string> GetReceiverConfig(
-            HttpRequest request,
+        /// <returns>
+        /// The configured WebHook secret key. <see langword="null"/> if the configuration value does not exist.
+        /// </returns>
+        protected virtual string GetSecretKey(
+            string sectionKey,
             RouteData routeData,
-            string configurationName,
             int minLength,
             int maxLength)
         {
-            if (request == null)
+            // Look up configuration for this receiver and instance.
+            var secrets = GetSecretKeys(sectionKey, routeData);
+            if (secrets == null)
```

Comment:
>null [](start = 27, length = 4)

I think should should be a .Exists() check not a null check

Reply:
Are there more cases like this - where we're null checking the result of getting a config section?

---
In reply to: [150144519](https://github.com/aspnet/WebHooks/pull/172#discussion_r150144519) [](ancestors = 150144519)

Reply:
Good catch. There were two places where I was still `null`-checking `IConfigurationSection`s.